From d5d8c1de6eeb66071845c85687426d3695e1aa21 Mon Sep 17 00:00:00 2001
From: Joel Holdsworth <jholdsworth@nvidia.com>
Date: Wed, 15 May 2024 14:49:29 -0700
Subject: [PATCH] tests: Use gcab_test_get_filename to construct test path

Previously, the gcab_test_cabinet_error_no_folders_func test assumed that it
could write to /tmp/foo.cab . However, on non-UNIX systems e.g. Windows, this
is not possible. This patch corrects the issue by using gcab_test_get_filename
to construct a test file path.

Signed-off-by: Joel Holdsworth <jholdsworth@nvidia.com>
---
 tests/gcab-self-test.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/gcab-self-test.c b/tests/gcab-self-test.c
index bc487b3..942b4fd 100644
--- a/tests/gcab-self-test.c
+++ b/tests/gcab-self-test.c
@@ -177,6 +177,7 @@ static void
 gcab_test_cabinet_error_no_folders_func (void)
 {
     gboolean ret;
+    g_autofree gchar *fn = NULL;
     g_autoptr(GCabCabinet) cabinet = NULL;
     g_autoptr(GError) error = NULL;
     g_autoptr(GFile) file = NULL;
@@ -184,7 +185,8 @@ gcab_test_cabinet_error_no_folders_func (void)
 
     /* we can't write with no folders */
     cabinet = gcab_cabinet_new ();
-    file = g_file_new_for_path ("/tmp/foo.cab");
+    fn = gcab_test_get_filename ("foo.cab");
+    file = g_file_new_for_path (fn);
     op = G_OUTPUT_STREAM (g_file_replace (file, NULL, FALSE, 0, NULL, &error));
     g_assert_no_error (error);
     g_assert (op != NULL);
-- 
2.45.1

